### PR TITLE
Add unit tests for identity package

### PR DESCRIPTION
**Pull request by geetikabatra**
This PR adds unit tests to check if cluster spec identity Ref
is secret or not

Signed-off-by: geetikab@vmware.com <geetikab@vmware.com>
Co-authored-by: Ankita Swamy <swamyan@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Review by srm09 — COMMENTED** [review]
minor changes but otherwise lgtm

File: pkg/identity/identity_test.go [hunk 1]
```diff
@@ -54,7 +56,29 @@ var _ = Describe("GetCredentials", func() {
 	AfterEach(func() {
 		Expect(k8sclient.Delete(ctx, ns)).To(Succeed())
 	})
+	fmt.Println("with checking if cluster spec identity Ref secret")
```
Comment (Ankitasw) [inline pkg/identity/identity_test.go:59]: Do we need these `Println` statements or added by mistake?

Comment (geetikabatra) [reply pkg/identity/identity_test.go:59]: I have added them for more verbosity. Without this, searchability of the tests in terminal was technically impossible.

Comment (srm09) [reply pkg/identity/identity_test.go:59]: There are other ways of reporting the tests that were run. Highly suggest that having print statements in tests for reporting running isn't the best way to achieve it. Do consider removing this

Comment (geetikabatra) [reply pkg/identity/identity_test.go:59]: Sure! I will ponder a little more into it.

**Comment by geetikabatra** [comment]
@srm09 and @Ankitasw PTAL

File: pkg/identity/identity_test.go [hunk 2]
```diff
@@ -54,7 +55,29 @@ var _ = Describe("GetCredentials", func() {
 	AfterEach(func() {
 		Expect(k8sclient.Delete(ctx, ns)).To(Succeed())
 	})
+	klog.Infof("with checking if cluster spec identity Ref secret")
```
Comment (srm09) [inline pkg/identity/identity_test.go:58]: Still unclear around why there are log statements describing the tests that are being run. Like I mentioned before adding log/printf statements for the purpose of reporting is not the best way to move forward.

File: go.mod [hunk 3]
```diff
@@ -27,6 +27,7 @@ require (
 	k8s.io/client-go v0.23.0
 	k8s.io/cluster-bootstrap v0.23.0
 	k8s.io/component-base v0.23.0
+	k8s.io/klog v1.0.0
```
Comment (srm09) [inline go.mod:30]: We use the v2 version of klogwhich is already imported. This can be removed.

**Review by srm09 — CHANGES_REQUESTED** [review]
Could you address the comments and we can move ahead and merge the PR.

**Comment by geetikabatra** [comment]
Removing all the logging lines for now. This is ready to merge.

**Pull request merged**